### PR TITLE
feat(Shared): Migrate tests

### PR DESCRIPTION
**Pull request by kevinvangelder**
Re-recreate #837/#889 but move from UWP tests instead of Net46 tests to retain history and excluding troublesome submodule changes.

### CODE REVIEW

**Comment by kevinvangelder** [comment]
AppVeyor only ran 70 of 122 tests.

**Comment by rozele** [comment]
Sounds like one of the tests is crashing the test process.  Can we make sure all the tests run before merging this?

---
In reply to: [262384685](https://github.com/ReactWindows/react-native-windows/pull/899#issuecomment-262384685) [](ancestors = 262384685)

**Comment by kevinvangelder** [comment]
Well, it's only discovering 70 of 122 tests, so it may be trying to use a cached binary or something. Matt said he'd take a look at it.

**Comment by kevinvangelder** [comment]
Update: Matt is contacting AppVeyor via his paid support channel to try to get this resolved.

**Comment by IlyaFinkelshteyn** [comment]
Hi Matt, Kevin and Eric, please confirm that to reproduce the issue, I need to fork https://github.com/infinitered/react-native-windows/ and run AppVeyor build against `rerecreate-move-tests` branch, right?

--ilya.

**Comment by kevinvangelder** [comment]
@IlyaFinkelshteyn Yes please.

**Comment by IlyaFinkelshteyn** [comment]
As I understand you can locally run all 120 tests. Are they all in the same test assembly? To run all 120 test manually locally do you still call `nunit3-console <local_path>\ReactNative.Net46.Tests.dll` or you have to call `nunit3-console' against multiple assemblies?
Other words do you think the problem is that AppVeyor does not discover all test assemblies properly or it does not run all test it expected to run fro, single assembly?

Meanwhile I am looking into your changes and doing repro, but your input would be great :)

**Comment by kevinvangelder** [comment]
@IlyaFinkelshteyn I ran just the `ReactNative.Net46.Tests.dll` through the `nunit3-console` for it to run those 120 tests, so I think the problem is that AppVeyor doesn't run the expected tests in the single assembly.

**Comment by IlyaFinkelshteyn** [comment]
Here is what I observe:
 - I forked `infinitered/react-native-windows` and did run AppVeyor build against `rerecreate-move-tests` branch. My results are [Test Count: 112, Passed: 108, Failed: 0, Inconclusive: 0, Skipped: 4](https://ci.appveyor.com/project/IlyaFinkelshteyn/react-native-windows/build/1.0.3#L1737). 

 - I compared [my results](https://ci.appveyor.com/project/IlyaFinkelshteyn/react-native-windows/build/tests) with [problem test run](https://ci.appveyor.com/project/rozele/react-native-windows/build/1.0.1330/tests)  and see that missed test are (at least) from the following classes:
   - `NativeModuleBaseTests`
   - `NativeModuleRegistryTests`
   - `PromiseTests`

 - What is special about those classes is that they all were 
   - moved to different folder
   - migrated from VS unit test framework to nunit

I feel that root cause is very close. What is most weird is that I have no repro, and I do not want to spoil your build with my PRs. 
Random idea - can you try the same PR, but with removing cache settings from `appveyor.yml`?

--ilya.

**Comment by kevinvangelder** [comment]
@IlyaFinkelshteyn Nope, as you can see in the above results we're still only getting 70 tests discovered.

**Comment by IlyaFinkelshteyn** [comment]
Can you please do the following sanity test: Create new AppVeyor project using `infinitered/react-native-windows` and set default branch to `rerecreate-move-tests`. Will all test be discovered as it happened for me?

**Comment by kevinvangelder** [comment]
Is it possible this failing clone is at fault and it's falling back to a previous clone? https://ci.appveyor.com/project/rozele/react-native-windows/build/1.0.1336#L7

**Comment by kevinvangelder** [comment]
@IlyaFinkelshteyn Mine found and ran all 112. https://ci.appveyor.com/project/kevinvangelder/react-native-windows

Any idea why `FETCH_HEAD` works on mine but not on Rozele's AppVeyor?

**Comment by IlyaFinkelshteyn** [comment]
@kevinvangelder I am trying to make sense of it. Meanwhile I have to say that PowerShell script does not fail and therefore build continues to run happily, if last command exit code is good. You have either replace script with individual commanda, or check for error after every (or most important commands) with something like `if (!$?) {throw}` or `IF ($LASTEXITCODE -ne 0){$host.SetShouldExit($LASTEXITCODE)}`.

**Comment by kevinvangelder** [comment]
@IlyaFinkelshteyn Ahha! The `FETCH_HEAD` code is only run when coming from a pull request. In our test cases, we're testing directly on a branch, so `"git checkout -qf " + $Env:APPVEYOR_REPO_COMMIT);` is run instead.

**Comment by kevinvangelder** [comment]
@IlyaFinkelshteyn Yes, we could probably do that, but it won't fix the fact that the git commands don't work as expected.

**Comment by IlyaFinkelshteyn** [comment]
@kevinvangelder here is working  working PowerShell based script:
```
clone_script:
- ps: git clone -q $("--branch=" + $Env:APPVEYOR_REPO_BRANCH) $("https://github.com/" + $Env:APPVEYOR_REPO_NAME + ".git") $Env:APPVEYOR_BUILD_FOLDER
- ps: if (!$Env:APPVEYOR_PULL_REQUEST_NUMBER) {$("git checkout -qf " + $Env:APPVEYOR_REPO_COMMIT)}
- ps: if ($Env:APPVEYOR_PULL_REQUEST_NUMBER) {git fetch -q origin +refs/pull/$($Env:APPVEYOR_PULL_REQUEST_NUMBER)/merge; git checkout -qf FETCH_HEAD}
- ps: git submodule update -q --init --recursive
```

Thanks!
Ilya.

File: ReactWindows/ReactNative.Shared.Tests/Modules/Network/NetworkingModuleTests.cs [hunk 1]
```diff
@@ -103,7 +113,11 @@ public void NetworkingModule_Request_Content_String()
             var waitHandle = new AutoResetEvent(false);
             var module = CreateNetworkingModule(new MockHttpClient(request =>
                 {
+#if WINDOWS_UWP
                     var body = request.Content.ReadAsStringAsync().GetResults();
+#else
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared.Tests/Modules/Network/NetworkingModuleTests.cs]: I'm surprised this works on UWP without an await...

Comment (kevinvangelder) [inline ReactWindows/ReactNative.Shared.Tests/Modules/Network/NetworkingModuleTests.cs]: Would you like me to add one?

Comment (rozele) [inline ReactWindows/ReactNative.Shared.Tests/Modules/Network/NetworkingModuleTests.cs]: No leave it for now, can you file an issue to investigate the UWP tests that use GetResults() though?

---
In reply to: [91598112](https://github.com/ReactWindows/react-native-windows/pull/899#discussion_r91598112) [](ancestors = 91598112)

Comment (rozele) [inline ReactWindows/ReactNative.Shared.Tests/Modules/Network/NetworkingModuleTests.cs]: Actually, I will.

---
In reply to: [91630290](https://github.com/ReactWindows/react-native-windows/pull/899#discussion_r91630290) [](ancestors = 91630290,91598112)

**Pull request merged**